flang/lib/Semantics/resolve-directives.cpp

```diff
@@ -1034,61 +1034,53 @@ void AccAttributeVisitor::AddRoutineInfoToSymbol(
     Symbol &symbol, const parser::OpenACCRoutineConstruct &x) {
```

Comment:
The code in this routine was previously wrong according to how code was generated for the decls later. Notice the device type clause just takes the first device and throws away the rest.